File: Week 6/caesar.py
from cs50 import get_string
import sys

if len(sys.argv) != 2:
    print("Usage: python caesar.py k")
    exit(1)

key = sys.argv[1]

print("plaintext: ", end="")
text = get_string("")
ctext = ""

for i in range(len(text)):
    if not text[i].isalpha():
        cText += text[i]
    elif text[i].isupper():
        cText += chr(((ord(text[i]) - 65 + int(key)) % 26) + 65)
    else:
        cText += chr(((ord(text[i]) - 97 + int(key)) % 26) + 97)

print("ciphertext: ", end="")
print(cText)

exit(0)

File: Week 2/caesar.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <stdlib.h>

int main(int argc, string argv[])
{
    if (argc != 2)
    {
        printf("Usage: ./caesar key\n");
        return 1;
    }

    for (int i = 0; i < strlen(argv[1]); i++)
    {
        if (argv[1][i] < 48 || argv[1][i] > 57) // key'in sayı kontrolü
        {
            printf("Usage: ./caesar key\n");
            return 1;
        }
    }
    int key;
    key = atoi(argv[1]);

    printf("plaintext: ");
    string s = get_string("");

    for (int n = 0; n < strlen(s); n++)
    {
        if (isalpha(s[n]) == 0)
        {
            s[n] = s[n];
        }
        else if (isupper(s[n]))
        {
            s[n] = (((s[n] - 65) + key) % 26) + 65;
        }
        else
        {
            s[n] = (((s[n] - 97) + key) % 26) + 97;
        }
    }
    printf("ciphertext: ");
    printf("%s\n", s);

    return 0;
}
